sympy/printing/pretty/tests/test_pretty.py

```diff
@@ -5553,9 +5553,9 @@ def test_issue_7180():
 
 
 def test_pretty_Complement():
-    assert pretty(S.Reals - S.Naturals) == '(-oo, oo) \ Naturals()'
+    assert pretty(S.Reals - S.Naturals) == 'Interval(-oo, oo, True, True) \ S.Naturals'
```

Comment:
Damn, this can be a problem. The pretty printer should print as closest as possible to the traditional mathematical printing way.

Maybe allow the pretty printer to use the former printer? It should not default to `str( )`.